test/unit/host_test.rb

```diff
@@ -1854,6 +1854,11 @@ def to_managed!
     assert_equal [pclass.name], host.individual_puppetclasses.map(&:name)
   end
 
+  test "host created with config group updates the correct count" do
```

Comment:
A corresponding Hostgroup test would be useful I think, since the logic is partly duplicated.